internal/gatewayapi/route.go

```diff
@@ -250,13 +251,19 @@ func (t *Translator) processHTTPRouteRules(httpRoute *HTTPRouteContext, parentRe
 					))
 					processDestinationError = err
 				}
-				continue
+				// Important: create a DestinationSetting with no endpoints to represent the invalid backendRef.
+				// This ensures requests are routed correctly between valid backends and the synthetic
+				// invalid-backend-cluster based on their respective weights.
+				ds = &ir.DestinationSetting{
```

Comment:
can we move this logic into `processDestination` which is responsible for `ds` generation